docs/agent-reference/how-to-build-agent-code.md

```diff
@@ -128,3 +136,41 @@ popd > /dev/null
 ```
 
 **Note** If the Device Update Agent was built as a daemon, the install targets will install and register the Device Update Agent as a daemon.
+
+## Run Device Update Agent
+
+Run Device Update Agent by following these [instructions](./how-to-run-agent.md)
+
+# Integrate the Device Update agent in your existing application or solution
+
+## Pre-concepts
+
+Before integrating the Device Update agent in your existing application or solution review the below concepts.
+
+* Learn how the Device Update service will communicate with the device client using IoT Hub Plug and Play properties to orchestrate over-the-air update actions from [here](https://github.com/Azure/iot-hub-device-update/blob/main/src/agent/adu_core_interface/src/agent_workflow.c).
```

Comment:
>agent_workflow.c) [](http://example.com/codeflow?start=262&length=17)

to be replaced by the diagrams Nox circulated recently,  right?

Reply:
Yes, the links we fine last week. I will double check what is wrong with them today. New images are part of this PR if you want to take a look.

Reply:
I see them, thanks Val!